capten/agent/pkg/agent/agent_sync_app.go

```diff
@@ -0,0 +1,149 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gocql/gocql"
+	"github.com/kube-tarian/kad/capten/agent/pkg/agentpb"
+	"gopkg.in/yaml.v2"
+)
+
+func (a Agent) syncApp(ctx context.Context, request *agentpb.SyncAppRequest) error {
+
+	var appConfig AppConfig
+	if err := yaml.Unmarshal(request.Payload, &appConfig); err != nil {
+		a.log.Errorf("could not unmarshal appConfig yaml: %v", err)
+		return err
+	}
+
+	// Note: We are creating keyspace and table when needed
+	if err := a.store.Session().Query(fmt.Sprintf(createKeyspaceCQL, "1")).Exec(); err != nil {
+		a.log.Errorf("could not create keyspace, err: %v", err)
+		return err
+	}
+
+	if err := a.store.Session().Query(createTableQuery).Exec(); err != nil {
+		a.log.Errorf("could not create table, err: %v", err)
+		return err
+	}
+
+	if err := insert(a.store.Session(), appConfig); err != nil {
+		a.log.Errorf("could not insert, err: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+// Todo: These structs along with proto file should be a part of common package
+
+type LaunchUIConfig struct {
+	RedirectURL string `yaml:"RedirectURL"`
+}
+
+type Override struct {
+	LaunchUIConfig LaunchUIConfig `yaml:"LaunchUIConfig"`
+	LaunchUIValues map[string]any `yaml:"LaunchUIValues"`
+	Values         map[string]any `yaml:"Values"`
+}
+
+type AppConfig struct {
+	Name                string   `yaml:"Name"`
+	ChartName           string   `yaml:"ChartName"`
+	RepoName            string   `yaml:"RepoName"`
+	RepoURL             string   `yaml:"RepoURL"`
+	Namespace           string   `yaml:"Namespace"`
+	ReleaseName         string   `yaml:"ReleaseName"`
+	Version             string   `yaml:"Version"`
+	Override            Override `yaml:"Override"` // this can be marshled as json and stored as string
+	CreateNamespace     bool     `yaml:"CreateNamespace"`
+	PrivilegedNamespace bool     `yaml:"PrivilegedNamespace"`
+}
+
+const createKeyspaceCQL = "CREATE KEYSPACE IF NOT EXISTS Apps WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : %s } AND DURABLE_WRITES = true"
```

Comment:
these queries not belong to agent synch, as this table used to fetch application config, update it etc..

Move this to new package "app"
--> This package handle app DB table.
-> provide syncapp, get with query, update, delete etc..
For now only implement methods needed for syncapp agent API

Reply:
Cassandra connection initialization also move to new app package

Reply:
done